metadata-transformer/src/main/java/no/unit/nva/metadata/service/MetadataService.java

```diff
@@ -113,6 +146,47 @@ private Model normalizeStatements(RepositoryResult<Statement> statements) {
         return model;
     }
 
+    private Optional<Statement> toBiboDoi(ValueFactory valueFactory, Statement statement) throws IOException,
+            InterruptedException {
+        String value = statement.getObject().stringValue();
+        String doi = isShortDoi(value) ? fetchDoiUriFromShortDoi(value).orElse(null)
+                : DOI_PREFIX + value.substring(value.indexOf(DOI_FIRST_PART));
+        IRI property = valueFactory.createIRI(BIBO_DOI);
+        return Optional.ofNullable(valueFactory.createStatement(statement.getSubject(),
```

Comment:
The code is quite dense. I would suggest to assign the object to a variable and the statement to another variable.

Reply:
I don't like local creating local variables that can be inlined; I might prefer just adding a getDoi method instead. But then again, this method is pretty good; I doubt I'd think twice about it if not for the combination of .orElse() within a ternary operator and valueFactory.createStatement being a bit of a clunky method. For me, it's probably just a matter of getting more used to ternary operators; although my person preference is for them to only be used in really simple scenarios.

Reply:
Unfortunately, the author of the code is quite dense. I have corrected the mistake and hopefully it is more readable now.